File: python/array/leetcode/shuffle_array.py
# 打乱数组
import random
from copy import copy
from typing import List


class Solution:

    def __init__(self, nums: List[int]):
        self.origin = copy(nums)
        self.nums = nums

    def reset(self) -> List[int]:
        return self.origin

    def shuffle(self) -> List[int]:
        for i in range(len(self.nums)):
            x = random.randint(0, len(self.nums) - 1)
            self.nums[i], self.nums[x] = self.nums[x], self.nums[i]
        return self.nums

File: go/array/leetcode/shuffle_array.go
package main

import (
	"math/rand"
)

//打乱数组

type Solution struct {
	nums []int
}

func Constructor(nums []int) Solution {
	return Solution{nums}
}

func (this *Solution) Reset() []int {
	return this.nums
}

func (this *Solution) Shuffle() []int {
	data := make([]int, len(this.nums))
	copy(data, this.nums)
	for i := 0; i < len(data); i++ {
		j := rand.Intn(len(data))
		data[i], data[j] = data[j], data[i]
	}
	return data
}
